modules/ROOT/pages/provisioning-kubevirt.adoc

```diff
@@ -0,0 +1,95 @@
+= Provisioning Fedora CoreOS on KubeVirt
+
+This guide shows how to provision new Fedora CoreOS (FCOS) nodes on any KubeVirt-enabled Kubernetes cluster.
+
+== Prerequisites
+
+Before provisioning an FCOS machine, you must have an Ignition configuration file containing your customizations. If you do not have one, see xref:producing-ign.adoc[Producing an Ignition File].
+
+You also need to have access to a Kubernetes environment with https://kubevirt.io/user-guide/operations/installation/[KubeVirt] installed.
+
+== Referencing the KubeVirt Image
+
+Fedora CoreOS is designed to be updated automatically, with different schedules per stream.
+
+The image for each stream can directly be referenced from the official registry:
+
+----
+quay.io/fedora/fedora-coreos-kubevirt
+----
+
+== Launching a VM instance
+
+Now that you have an image it can be used in your VMI definitions. The following example will use the image from the stable stream.
+
+You'll also need the Ignition config.
+Here the Ignition config is exposed to the VMI via a Kubernetes Secret.
+Learn about various ways to expose userdata to VMIs in the https://kubevirt.io/user-guide/virtual_machines/startup_scripts/#startup-scripts[KubeVirt user guide].
+
+NOTE: You must edit the following commands per your container management system. For example, you must replace `kubectl` with `oc` if you use Openshift.
+
+.Creating the secret
+[source, bash]
+----
+kubectl create secret generic ignition-payload --from-file=userdata=example.ign
+----
+
+.Launching a VM instance referencing the secret
+[source, bash]
+----
+cat << END > vmi.yaml
+---
+apiVersion: kubevirt.io/v1
+kind: VirtualMachineInstance
+metadata:
+  name: my-fcos
+spec:
+  domain:
+    devices:
+      disks:
+      - disk:
+          bus: virtio
+        name: containerdisk
+      - disk:
+          bus: virtio
+        name: cloudinitdisk
+      rng: {}
+    resources:
+      requests:
+        memory: 2048M
+  volumes:
+  - containerDisk:
+      image: quay.io/coreos/fedora-coreos:stable
+    name: containerdisk
+  - name: cloudinitdisk
+    cloudInitConfigDrive:
+      secretRef:
+        name: ignition-payload
+END
+kubectl create -f vmi.yaml
+----
+
+Now you should be able to SSH into the instance. If you didn't change the defaults, the
+username is `core`.
+
+.Accessing the VM instance using kubectl via ssh
+[source, bash]
+----
+kubectl virt ssh core@my-fcos
+----
+.Accessing the VM instance using virtctl via ssh
+[source, bash]
+----
+virtctl ssh core@my-fcos
+----
```

Comment:
would be nice if we could link to some docs about `virtctl` here, but I don't there really are any. Could you find some?

Reply:
I can only see https://kubevirt.io/user-guide/operations/virtctl_client_tool/ which is decent and will help to install virtctl for users who don't have it installed. You've probably already seen this option but there really isn't any other good option.

Reply:
That looks great. Let's use that.